src/Http/ServerRequest.php

```diff
@@ -517,6 +544,32 @@ public function is(array|string $type, mixed ...$args): bool
         return $this->_detectorCache[$type] = $this->_detectorCache[$type] ?? $this->_is($type, $args);
     }
 
+    /**
+     * Returns a piece of data from the Request converted to a given data type.
+     *
+     * Example: `asType('Int', $this->params, 'id', 0)` will return
+     * the 'id' parameter from the request converted to an integer.
+     *
+     * @param string $type The desired type to convert the data to. Example: 'Int', 'Bool', 'String'.
+     * @param object|array $source Source from which data should be extracted.
+     * @param mixed ...$args List of arguments
+     * @return mixed The requested data converted to the specified type.
+     * @throws \InvalidArgumentException If no detector has been set for the provided type.
+     */
+    protected function asType(string $type, array|object $source, mixed ...$args): mixed
```

Comment:
Is this something we would want to make extensible? I could see custom casters for enums or model resolvers. It could be interesting to have `$user = $this->request->asType('User', $id)` as this could work well with route parameters :shrug:

Reply:
That is a great idea. Should I change the accessibility to public to make it overridable?

Also, do you think a date converter like `$this->request->asType('DateTime', $this->query, 'start_date')` would make more sense in core code or userland code? I ask because date and time formats vary greatly from project-to-project, and I was lacking confidence in my date conversion code in the global function.

Reply:
> Also, do you think a date converter like $this->request->asType('DateTime', $this->query, 'start_date') would make more sense in core code or userland code?

I think we make it userland for the formatting reasons. If a common enough pattern comes up we can implement helpers to support the workflow. We can always expand API surface area in the future.